lib/classes/PluginManager.js

```diff
@@ -72,19 +78,23 @@ class PluginManager {
     events.forEach((event) => {
       const hooksForEvent = [];
       this.plugins.forEach((pluginInstance) => {
-        forEach(pluginInstance.hooks, (hook, hookKey) => {
-          if (hookKey === event) {
-            // use arrow fn. to pass the hook with options rather than calling it
-            hooksForEvent.push(() => hook());
-          }
-        });
+        // if a provider is given it should only add the hook when the plugins provider matches
+        // the services provider
+        if (!pluginInstance.provider || (pluginInstance.provider === this.provider)) {
+          forEach(pluginInstance.hooks, (hook, hookKey) => {
+            if (hookKey === event) {
+              // use arrow fn. to pass the hook with options rather than calling it
+              hooksForEvent.push(() => hook());
+            }
+          });
+        }
       });
       hooks = hooks.concat(hooksForEvent);
     });
 
     if (hooks.length === 0) {
       const errorMessage = `The command you entered was not found.
-     Did you spell it correctly?`;
+        Did you spell it correctly?`;
```

Comment:
Could you confirm that this message is printed correctly? otherwise could you switch to the convention I shared above?